### ISSUE TITLE
Separate workers

### ISSUE THREAD

**Issue by kafejo** [thefuntasty/Funtasty-CleanSwift-templates#14]
We use in our current projects workers that are "resource based". For example ArtistWorker, which defines all methods for store. This worker is then initialized with Store. The store has to conform ArtistWorkerProtocol. I found ArtistsWorker as unnecessary because it only contains store, there is no other use for that. 

Proposed solution: Use scene-based workers as a protocols that stores will be conforming to.

Example:
Lets say we have a scene List of artists. This scene will contain its worker (ListOfArtistsWorker) which defines what scene needs and acts as main worker (because interactor can have many workers).

``` swift
protocol ListOfArtistsWorker {
    func fetchAllArtists(callback callback: Result<[Artist]> -> ())
}
```

Then store can implement this (`ArtistMockupStore.swift`)

``` swift
// MARK: - ListOfArtistsWorker

extension ArtistMockupStore: ListOfArtistsWorker {
    func fetchAllArtists(callback callback: Result<[Artist]> -> ()) {
        callback(Result.Success(ArtistMockupStore.MockupArtists))
    }
}
```

The naming convention may be different. We can call it `__SceneName__`WorkerProtocol.

Full example [here](https://github.com/kafejo/MusicSheet)

**Comment by jakubknejzlik** [comment]
@kafejo please take a look at PR #17, is this the way you meant it?

**Comment by kafejo** [comment]
Resolved in #17
